extensions/wikia/SpecialDiscussions/SpecialDiscussionsController.class.php

```diff
@@ -0,0 +1,165 @@
+
+		$this->wg->Out->setPageTitle( wfMessage( 'discussions-pagetitle' )->escaped() );
+
+		$editToken = $this->getVal( self::EDIT_TOKEN );
+		if ( !empty( $editToken ) ) {
+			$this->activateDiscussions();
+		}
+
+		$this->setIndexOutput();
+	}
+
+	public function inputForm() {
+		$this->checkAccess();
+
+		$this->response->setValues(
+			[
+				'discussionsInactiveMessage' => wfMessage( 'discussions-not-active' )->escaped(),
+				'activateDiscussions' => wfMessage( 'discussions-activate' )->escaped(),
+				'editToken' => $this->getUser()->getEditToken(),
+			]
+		);
+	}
+
+	public function discussionsLink() {
+		$this->checkAccess();
+
+		$this->response->setValues(
+			[
+				'discussionsActiveMessage' => wfMessage( 'discussions-active' )->escaped(),
+				'discussionsLink' => $this->getDiscussionsLink(),
+				'discussionsLinkCaption' => wfMessage( 'discussions-navigate' )->escaped(),
+			]
+		);
+	}
+
+	private function setIndexOutput() {
+		if ( $this->getIsDiscussionsActive() ) {
+			$this->response->setVal(
+				'discussionsLink',
+				$this->sendSelfRequest(
+					'discussionsLink',
+					[
+						'discussionsLink' => $this->getDiscussionsLink(),
+					]
+				)
+			);
+		} else {
+			$this->response->setVal(
+				'inputForm',
+				$this->sendSelfRequest(
+					'inputForm',
+					[
+						'activateDiscussions' => $this->getActivateDiscussionsMessage(),
+					]
+				)
+			);
+		}
+	}
+
+	private function activateDiscussions() {
+		$this->assertValidPostRequest( $this->request, $this->getUser() );
+
+		$site = new \Swagger\Client\Discussion\Models\Site(
+			[
+				'id' => $this->siteId,
+				'languageCode' => F::app()->wg->ContLang->getCode(),
+			]
+		);
+
+		try {
+			$this->getDiscussionSitesApi()->createSite( $site, F::app()->wg->TheSchwartzSecretToken );
+		} catch ( ApiException $e ) {
+			$this->logger->error(
+				'Creating site caused an error',
+				[
+					'siteId' => $this->siteId,
+					'error' => $e->getMessage(),
+				]
+			);
+			throw new ErrorPageError( 'unknown-error', 'discussions-activate-error' );
+		}
+	}
+
+	private function getIsDiscussionsActive() {
+		try {
+			$this->getDiscussionSitesApi()->getSite( $this->siteId );
+			return true;
+		} catch ( ApiException $e ) {
+			$this->logger->debug( 'Getting site caused an error',
+				[
+					'siteId' => $this->siteId,
+					'error' => $e->getMessage(),
+				] );
+		}
+		return false;
+	}
+
+	private function getDiscussionsLink() {
+		return "/d/f";
+	}
+
+	private function getActivateDiscussionsMessage() {
+		return wfMessage( 'discussions-activate' )->escaped();
+	}
+
+	private function getDiscussionSitesApi() {
+		/** @var ApiProvider $apiProvider */
+		$apiProvider = Injector::getInjector()->get( ApiProvider::class );
+		/** @var SitesApi $api */
+		$api = $apiProvider->getApi( self::SERVICE_NAME, SitesApi::class );
+		$api->getApiClient()->getConfig()->setCurlTimeout( self::TIMEOUT );
+
+		return $api;
+	}
+
+	private function checkAccess() {
```

Comment:
Since this operates in the same way as `assertValidPostRequest`, maybe this should be `assertCanAccess`?